apps/solr.go

```diff
@@ -52,3 +53,76 @@ func (r MetricsReceiverSolr) Pipelines() []otel.Pipeline {
 func init() {
 	confgenerator.MetricsReceiverTypes.RegisterType(func() confgenerator.Component { return &MetricsReceiverSolr{} })
 }
+
+type LoggingProcessorSolrSystem struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorSolrSystem) Type() string {
+	return "solr_system"
+}
+
+func (p LoggingProcessorSolrSystem) Components(tag string, uid string) []fluentbit.Component {
+	c := confgenerator.LoggingProcessorParseMultilineRegex{
+		LoggingProcessorParseRegexComplex: confgenerator.LoggingProcessorParseRegexComplex{
+			Parsers: []confgenerator.RegexParser{
+				{
+					// Sample line: 2022-01-06 04:16:08.794 INFO  (qtp1489933928-64) [   x:gettingstarted] o.a.s.c.S.Request [gettingstarted]  webapp=/solr path=/get params={q=*:*&_=1641440398872} status=0 QTime=2
+					Regex: `^(?<timestamp>\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3,6})\s(?<severity>[A-z]+)\s{2}\((?<thread>[^\)]+)\)\s\[c?:?(?<collection>[^\s]*)\ss?:?(?<shard>[^\s]*)\sr?:?(?<replica>[^\s]*)\sx?:?(?<core>[^\]]*)\]\s(?<source>[^\s]+)\s(?<message>(?:(?!\s\=\>)[\s\S])+)\s?=?>?(?<exception>[\s\S]*)`,
+					Parser: confgenerator.ParserShared{
+						TimeKey:    "timestamp",
+						TimeFormat: "%Y-%m-%d %H:%M:%S.%L",
+					},
+				},
+			},
+		},
+		Rules: []confgenerator.MultilineRule{
+			{
+				StateName: "start_state",
+				NextState: "cont",
+				Regex:     `\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3}\s[A-z]+\s{2}`,
+			},
+			{
+				StateName: "cont",
+				NextState: "cont",
+				Regex:     `^(?!\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3}\s[A-z]+\s{2})`,
+			},
+		},
+	}.Components(tag, uid)
+
+	// https://solr.apache.org/guide/6_6/configuring-logging.html
+	c = append(c,
+		fluentbit.TranslationComponents(tag, "level", "logging.googleapis.com/severity", true,
+			[]struct{ SrcVal, DestVal string }{
+				{"FINEST", "DEBUG"},
+				{"FINE", "DEBUG"},
+				{"CONFIG", "ERROR"},
+				{"INFO", "INFO"},
+				{"WARN", "WARNING"},
+				{"SEVERE", "CRITICAL"},
+			},
+		)...,
+	)
+	return c
+}
+
+type SystemLoggingReceiverSolr struct {
```

Comment:
@jefferbrecht I will change this. There is a Tomcat follows this format, but ultimately you're right. Renaming this would be in line with most of the other receivers.